Feature#57 improve url for users

EduJimenez96 opened the pull request:
# Description
 - What?: I've changed the url that shows the user data.
 - Why?: In order to prevent users to see another users' id in the url
 
 # Testing
 None

  ## Screenshots 
Url moddified
![url](https://user-images.githubusercontent.com/69607052/107396547-78a24700-6ac3-11eb-9ae4-3012717d1ef3.png)

EduJimenez96 committed:
change url 'user/id' into 'my-account'
Only remains to hide the id in the url

EduJimenez96 committed:
hide the user id from the url

YilKanda reviewed (dismissed):
LGTM!

JacquelineFM reviewed (dismissed):
Nice!

EduardoGHdez commented on config/routes.rb:
```diff
@@ -15,6 +15,8 @@
       only: [:create, :edit, :update]
   end
 
+  get 'my-account', to: 'users#show', as: 'my-account'
```
I think I mentioned this earlier, but, Would you consider using an `index` action instead of a `show`?

I mean, what if we just retrieve data from the logged user instead of querying the user by the passed params.

If what I say makes sense to you, but you don't know how to start. I'd be more than happy to help you with it.

EduardoGHdez replied:
In fact, it'ss what you are doing XD

The `name` passed via param it's not even used in the `show` method haha

EduardoGHdez replied:
I'd suggest to move it to an index, because `show` actions are supposed to receive an `id` via `url-params`

EduardoGHdez commented on app/views/shared/_sidebar.html.erb:
```diff
@@ -13,7 +13,7 @@
       <% if current_user.present? %>
         <span class="sidebar-item"><button id="modal-register" class="btn text-left">
-          <button id="modal-register" class="btn text-left"><%= link_to "Mi cuenta", user_path(current_user) %></button>
+          <button id="modal-register" class="btn text-left"><%= link_to "Mi cuenta", my_account_path(current_user) %></button>
```
What happens if you do not pass the `current_user` as param here?
Try it 😄 
I think it's not needed and it'll continue to work

```suggestion
          <button id="modal-register" class="btn text-left"><%= link_to "Mi cuenta", my_account_path %></button>
```

YilKanda commented on config/routes.rb:
```diff
@@ -15,6 +15,8 @@
       only: [:create, :edit, :update]
   end
 
+  get 'my-account', to: 'users#index', as: 'my-account'
+
```
Verify, since changing from users#show to users#index, is causing errors in the redirect. 

When creating a new user: 
![imagen](https://user-images.githubusercontent.com/12243331/107594139-d4a9c000-6bd6-11eb-9392-7564a1746c1c.png)

In the edit account window, click Cancel. 
![imagen](https://user-images.githubusercontent.com/12243331/107594148-dbd0ce00-6bd6-11eb-9895-0b0b358e2706.png)
In the edit account window, click Edit. 
![imagen](https://user-images.githubusercontent.com/12243331/107594166-e4290900-6bd6-11eb-9f77-96204c65a7e2.png)

EduJimenez96 replied:
Done

EduJimenez96 committed:
redirect to market places after login
Redirect to market places instead home#index in order to improve the user experience

YilKanda reviewed (approved):
LGTM!

EduardoGHdez commented on app/controllers/users_controller.rb:
```diff
@@ -16,12 +16,14 @@ def new
+    redirect_to market_places_path
   end
 
   def show
+  end
```
I think we can remove this path since is no longer needed

EduardoGHdez replied:
Do you mind removing the route definition as well? 😄

EduardoGHdez reviewed (approved):
Minor comment

But I think you can merge after solving it 🙌🏽 

Nicely done!
